src/transformers/training_args.py

```diff
@@ -1530,6 +1530,13 @@ class TrainingArguments:
         },
     )
 
+    average_tokens_across_devices: Optional[bool] = field(
+        default=False,
+        metadata={
+            "help": "Whether or not to average tokens across devices."
+        }
+    )
```

Comment:
During the `__post_init__` we call `setup_devices`. We can change `average_tokens_across_devices` value to `False` if the world size < 1 I think!

This then simplifies it earlier to just be `if self.args.average_tokens_across_devices`

Reply:
Thanks, already fixed it, please check.